contracts/proxy/transparent/TransparentUpgradeableProxy.sol

```diff
@@ -44,12 +46,11 @@ interface ITransparentUpgradeableProxy is IERC1967 {
  * Our recommendation is for the dedicated account to be an instance of the {ProxyAdmin} contract. If set up this way,
  * you should think of the `ProxyAdmin` instance as the real administrative interface of your proxy.
  *
- * WARNING: This contract does not inherit from {ITransparentUpgradeableProxy}, and the admin function is implicitly
- * implemented using a custom call-routing mechanism in `_fallback`. Consequently, the compiler will not produce an
- * ABI for this contract. Also, if you inherit from this contract and add additional functions, the compiler will not
- * check that there are no selector conflicts. A selector clash between any new function and the functions declared in
- * {ITransparentUpgradeableProxy} will be resolved in favor of the new one. This could render the admin operations
- * inaccessible, which could prevent upgradeability.
+ * WARNING: This contract does not inherit from {ITransparentUpgradeableProxy}, and the admin functions are
+ * implemented by functions that have signature designed to match the selector of the {ITransparentUpgradeableProxy}
+ * interface, but without any arguments. This allows use to decode the argument manually after the `ifAdmin` modifier
+ * has had a change to forward the call. This is done so that the argument decoding does not fail here in case the
+ * proxy and the implementation have "selector clash".
```

Comment:
```suggestion
 * WARNING: This contract does not inherit from {ITransparentUpgradeableProxy} because its functions are implemented
 * in this contract with alternative signatures designed to match the selectors of the {ITransparentUpgradeableProxy}
 * interface, but without any arguments. This allows manual argument decoding after the `ifAdmin` modifier
 * has had a chance to forward the call, so that the argument decoding does not fail here in case the
 * proxy and the implementation have clashing selectors.
```

Reply:
Ah, you replaced it first @frangio.
I'm solving but leaving it in case you think it's useful

Reply:
Yeah I thought this text was describing implementation detail too much. Please look at the new text.

Reply:
Looks good, only comment might be:

```
 * This is necessary to fully implement transparency without decoding *reverts* caused by selector clashes between the
```

"errors" sound very vague to me when the issue is an actual decoding revert, what do you think?